risc0/zkvm/platform/src/syscall.rs

```diff
@@ -245,12 +282,116 @@ pub unsafe fn sys_sha_buffer(
     unimplemented!()
 }
 
-#[inline(always)]
-pub unsafe fn sys_rand(recv_buf: *mut u32, words: usize) {
-    #[cfg(target_os = "zkvm")]
-    {
-        sys_io(recv_buf, words, ptr::null_mut(), 0, SENDRECV_CHANNEL_RANDOM);
+pub unsafe extern "C" fn sys_rand(recv_buf: *mut u32, words: usize) {
+    syscall_0(nr::SYS_RANDOM, recv_buf, words);
+}
+
+pub unsafe extern "C" fn sys_panic(msg_ptr: *const u8, len: usize) -> ! {
+    syscall_2(nr::SYS_PANIC, null_mut(), 0, msg_ptr as u32, len as u32);
+    unreachable!()
+}
+
+pub unsafe extern "C" fn sys_log(msg_ptr: *const u8, len: usize) {
+    syscall_2(nr::SYS_LOG, null_mut(), 0, msg_ptr as u32, len as u32);
+}
+
+pub extern "C" fn sys_cycle_count() -> usize {
+    unsafe {
+        let Return(a0, _) = syscall_0(nr::SYS_CYCLE_COUNT, null_mut(), 0);
+        a0 as usize
     }
-    #[cfg(not(target_os = "zkvm"))]
-    unimplemented!()
+}
+
+/// Returns the number of bytes currently available for reading on the
+/// given file descriptor.
+pub unsafe extern "C" fn sys_read_avail(fd: u32) -> usize {
+    let Return(navail, _) = syscall_1(nr::SYS_READ_AVAIL, null_mut(), 0, fd);
+    navail as usize
+}
+
+/// Reads the given number of bytes into the given buffer, posix-style.  Returns
+/// the number of bytes actually read.  On end of file, returns 0.
```

Comment:
When the caller asks for `nrequested` bytes, but the number of read bytes is fewer does that mean the fd is closed? (My guess would be yes, but that's not clear in this comment)

Reply:
Not necessarily; it has standard posix semantics.  Clarified comment.